ddtrace/contrib/internal/openai/utils.py

```diff
@@ -429,25 +479,50 @@ def _compute_completion_tokens(completions_or_messages, model_name):
     return estimated, num_completion_tokens
 
 
-def _tag_tool_calls(integration, span, tool_calls, choice_idx):
+def _tag_tool_calls(integration, span, tool_calls, choice_idx=None):
     # type: (...) -> None
     """
-    Tagging logic if function_call or tool_calls are provided in the chat response.
+    Tagging logic if function_call, tool_calls in the chat response, or tools are provided in the responses.
     Notes:
         - since function calls are deprecated and will be replaced with tool calls, apply the same tagging logic/schema.
         - streamed responses are processed and collected as dictionaries rather than objects,
           so we need to handle both ways of accessing values.
     """
+    # Handle response tools
+    if hasattr(tool_calls, "tools") and tool_calls.tools != []:
+        response_tools = []
+        for tool in tool_calls.tools:
+            if not (hasattr(tool, "type") or hasattr(tool, "name")):
+                continue
+
+            tool_dict = {}
+            if hasattr(tool, "type"):
+                tool_dict["type"] = getattr(tool, "type")
+            if hasattr(tool, "name"):
+                tool_dict["name"] = getattr(tool, "name")
+
+            if tool_dict:
+                response_tools.append(tool_dict)
+        # Only set the tag if we have valid tools to report
+        if response_tools:
+            span.set_tag("openai.response.tools", response_tools)
+        return
```

Comment:
If we're returning here, we won't be able to hit line 512. Is this what we want? Do we not want to tag the arguments as well?